mediathread/assetmgr/views.py

```diff
@@ -324,6 +328,61 @@ def post(self, request):
             return HttpResponseRedirect(asset_url)
 
 
+class UploadedAssetCreateView(LoggedInCourseMixin, View):
+    """
+    View for creating an Asset via an uploaded media object.
+    """
+    http_method_names = ['post']
+
+    def dispatch(self, request, *args, **kwargs):
+        r = super().dispatch(request, *args, **kwargs)
+
+        if not (request.user.is_staff or
+                request.course.is_faculty(request.user)):
+            raise PermissionDenied
+
+        return r
+
+    def post(self, request, *args, **kwargs):
+        if (not request.POST.get('title')) or \
+           (not request.POST.get('url')):
+            return HttpResponseBadRequest(
+                'Title and URL are required to make an asset.')
+
+        title = request.POST.get('title').strip()
+        url = request.POST.get('url')
+
+        asset = Asset.objects.create(
+            course=request.course, title=title, author=request.user)
+
+        label = 'image'
+        if url.endswith('.pdf'):
+            label = 'pdf'
+            # Dimensions are not needed for PDF display.
+            width = None
+            height = None
+        else:
+            width = request.POST.get('width')
+            height = request.POST.get('height')
+
+        Source.objects.create(
+            asset=asset, url=url,
+            primary=True,
+            width=width, height=height,
+            label=label)
+
+        asset_url = reverse('asset-view', args=[
```

Comment:
Before returning, a global annotation needs to be created to ensure the asset shows up in the user's collection.

`asset.global_annotation(user, True)`